tests/object_sync_test.go

```diff
@@ -1400,16 +1433,25 @@ func makeCustomVarTestData(t *testing.T) []*CustomVarTestData {
 			"b": `[true,false,null]`,
 			"c": `{"a":"foo","b":"bar","c":42}`,
 		},
-		VarsFlat: map[string]string{
-			"a":    `-13.37`,
-			"b[0]": `true`,
-			"b[1]": `false`,
-			"b[2]": `null`,
-			"c.a":  "foo",
-			"c.b":  `bar`,
-			"c.c":  `42`,
+		VarsFlat: map[string]sql.NullString{
+			"a":    toDBString(`-13.37`),
+			"b[0]": toDBString(`true`),
+			"b[1]": toDBString(`false`),
+			"b[2]": toDBString("null"),
+			"c.a":  toDBString("foo"),
+			"c.b":  toDBString(`bar`),
+			"c.c":  toDBString(`42`),
 		},
 	})
 
 	return data
 }
+
+func toDBString(str string) sql.NullString {
+	result := sql.NullString{String: str, Valid: true}
+	if str == "" {
+		result.Valid = false
+	}
+
+	return result
+}
```

Comment:
If my intuition in my comment on `expected[k] = toDBString(v)` is correct, it should be possible to remove the `str == ""` case here as I don't think there's a situation in the tests where we would want this. This would then allow using `toDBString(str)` as a shortcut for `sql.NullString{String: str, Valid: true}` (whereas `NULL` can be achieved with `types.String{}`).